java/ql/src/experimental/Security/CWE/CWE-600/UncaughtServletException.ql

```diff
@@ -0,0 +1,79 @@
+/**
+ * @name Uncaught Servlet Exception
+ * @description Uncaught exceptions in a servlet could leave a system in a vulnerable state, possibly resulting in denial-of-service attacks or the exposure of sensitive information disclosed in stack traces.
+ * @kind path-problem
+ * @id java/uncaught-servlet-exception
+ * @tags security
+ *       external/cwe-600
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.frameworks.Servlets
+import DataFlow::PathGraph
+
+/** The type `java.io.IOException`. */
+class IOException extends RefType {
+  IOException() { this.hasQualifiedName("java.io", "IOException") }
+}
+
+/** Check whether a given exception type is caught. */
+private predicate catchesEx(TryStmt t, RefType exType) {
+  exists(CatchClause cc, LocalVariableDeclExpr v |
+    t.getACatchClause() = cc and
+    cc.getVariable() = v and
+    v
+        .getType()
+        .(RefType)
+        .getASubtype*()
+        .hasQualifiedName(exType.getPackage().getName(), exType.getName()) //Detect the case that a subclass exception is thrown but its parent class is declared in the catch clause.
```

Comment:
Isn't this essentially
`v.getType().(RefType).getASubType*() = exType`
?

Reply:
Thanks @intrigus-lgtm . Yes, it's essentially the same but the suggested format is both concise and easy to read. I have tried something similar but got an invalid grammar error then I used the lengthy format.

I've committed the suggested change.